scripts/build-plugins-report.js

```diff
@@ -0,0 +1,183 @@
+#!/usr/bin/env node
+/*
+ * Copyright 2023 The Backstage Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+const fs = require('fs-extra');
+const path = require('path');
+const { execFile: execFileCb } = require('child_process');
+const { promisify } = require('util');
+
+const execFile = promisify(execFileCb);
+
+const rootDirectory = path.resolve(__dirname, '..');
+const pluginsDirectory = path.resolve(rootDirectory, 'plugins');
+
+async function run(command, ...args) {
+  const { stdout, stderr } = await execFile(command, args, {
+    cwd: rootDirectory,
+  });
+
+  if (stderr) {
+    console.error(stderr);
+  }
+
+  return stdout.trim();
+}
+
+function findLatestValidCommit(commits, directoryPath) {
```

Comment:
I think it'd be useful to filter out shared commits too but let's do that in a followup. I could hack on that later today unless you want to call dibs 😁

Reply:
Amazing, go for it, and thanks for all your ideas 🙌🏻